xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -0,0 +1,355 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use crate::{new_executor, worst_case_holding, XcmCallOf};
+use codec::Encode;
+use frame_benchmarking::{benchmarks, BenchmarkError};
+use frame_support::dispatch::GetDispatchInfo;
+use sp_std::vec;
+use xcm::{
+	latest::{prelude::*, MultiAssets},
+	DoubleEncoded,
+};
+
+const MAX_ASSETS: u32 = 100;
+
+benchmarks! {
+	query_holding {
+		let holding = worst_case_holding();
+
+		let mut executor = new_executor::<T>(Default::default());
+		executor.holding = holding;
+
+		let instruction = Instruction::<XcmCallOf<T>>::QueryHolding {
+			query_id: Default::default(),
+			dest: T::valid_destination()?,
+			assets: Wild(WildMultiAsset::All), // TODO is worst case filter?
```

Comment:
Actually no, this is in fact the best case filter -- while processing the `QueryHolding` instruction, the executor calls the [`Assets::min` function](https://github.com/paritytech/polkadot/blob/f26005b999acb8936bda66df10daa0d75c85ce62/xcm/xcm-executor/src/assets.rs#L397-L431), which looks to have the worst case that's based on the number of assets that you if you have a `MultiAssetFilter::Definite`. If instead you have a `WildNonFungible`, then it would depend upon the number of non-fungible asset classes that exist in holding.

Either way, I don't think there's a way to not break the weight calculation into separate cases based on the `assets` parameter, sadly.

Reply:
Fixed. Worst case as I can see is using `Definite(all_assets_explicitly)`.